### ISSUE TITLE
Crunchyroll authentication broke

### ISSUE THREAD

**Issue by kg** [streamlink/streamlink#3121]
## Bug Report

- [x] This is a bug report and I have read the contribution guidelines.

### Description

Crunchyroll authentication seems to have broken sometime in the last week. I successfully downloaded some episodes with my current rc/login a few days ago but now login fails. The actual CR website works fine, I'm logged in there.

### Expected / Actual behavior

Expected behavior: stream plays/saves correctly
Actual behavior:
```
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 369, in _create_api
    login = api.authenticate()
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 217, in authenticate
    data = self._api_call("authenticate", {"auth": self.auth}, schema=_login_schema)
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 173, in _api_call
    raise CrunchyrollAPIError(err_msg, err_code)
streamlink.plugin.crunchyroll.CrunchyrollAPIError: User could not be found.
```

### Reproduction steps / Explicit stream URLs to test

streamlink https://www.crunchyroll.com/rezero-starting-life-in-another-world-/episode-32-friend-796210 best -o rezero.mkv --crunchyroll-username=xxxx --crunchyroll-password=yyyy

### Log output

```
C:\Users\kg\Desktop>streamlink https://www.crunchyroll.com/rezero-starting-life-in-another-world-/episode-32-friend-796210 best -o rezero.mkv --loglevel debug
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.6.6
[cli][debug] Streamlink: 1.5.0
[cli][debug] Requests(2.24.0), Socks(1.7.1), Websocket(0.57.0)
[cli][info] Found matching plugin crunchyroll for URL https://www.crunchyroll.com/rezero-starting-life-in-another-world-/episode-32-friend-796210
[cli][debug] Plugin specific arguments:
[cli][debug]  --crunchyroll-username=<snip> (username)
[cli][debug]  --crunchyroll-password=******** (password)
[utils.l10n][debug] Language code: en_US
[plugin.crunchyroll][debug] Creating session with locale: en_US
[plugin.crunchyroll][debug] Session created with ID: cb4095f96be159e6cbae0373919a1a1f
[plugin.crunchyroll][debug] Using saved credentials
Traceback (most recent call last):
  File "runpy.py", line 193, in _run_module_as_main
  File "runpy.py", line 85, in _run_code
  File "C:\Users\kg\AppData\Local\Streamlink\bin\streamlink.exe\__main__.py", line 18, in <module>
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink_cli\main.py", line 1024, in main
    handle_url()
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink_cli\main.py", line 585, in handle_url
    streams = fetch_streams(plugin)
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink_cli\main.py", line 465, in fetch_streams
    sorting_excludes=args.stream_sorting_excludes)
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugin\plugin.py", line 317, in streams
    ostreams = self._get_streams()
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 306, in _get_streams
    api = self._create_api()
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 369, in _create_api
    login = api.authenticate()
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 217, in authenticate
    data = self._api_call("authenticate", {"auth": self.auth}, schema=_login_schema)
  File "C:\Users\kg\AppData\Local\Streamlink\pkgs\streamlink\plugins\crunchyroll.py", line 173, in _api_call
    raise CrunchyrollAPIError(err_msg, err_code)
streamlink.plugin.crunchyroll.CrunchyrollAPIError: User could not be found.
```

### Additional comments, screenshots, etc.

I tried manually passing a username/password using the command line arguments in addition to using saved credentials/the settings in my streamlinkrc, it doesn't seem to work either.

**Comment by gravyboat** [comment]
You probably need to purge your credentials: Use https://streamlink.github.io/cli.html#cmdoption-crunchyroll-purge-credentials

**Comment by kg** [comment]
> You probably need to purge your credentials: Use https://streamlink.github.io/cli.html#cmdoption-crunchyroll-purge-credentials

That did it, thanks. Might be worth putting that somewhere in the error message for this failure case if it's straightforward to do. I'm guessing the credentials just expire, and this isn't an API or server change?

**Comment by gravyboat** [comment]
@kg Yes the credentials just expire after some time. There was some discussion about noting this in the past but we'd have to parse the error message we get back from Crunchyroll and since that could change it could break things so it would be annoying for maintenance in the event they changed it.

**Comment by kg** [comment]
Using ```--crunchyroll-purge-credentials``` isn't working anymore:
```
error: Unable to validate API response: Unable to validate key 'auth': Type of None should be 'str' but is 'NoneType'
```
And sadly this seems to prevent it from flushing the bad cached credentials when you call it.

I tried deleting my ```plugin-cache.json``` and now it just gives me that 'unable to validate key' error all the time, which is unfortunate.

I have my cr username/password in my streamlinkrc, and I successfully downloaded some episodes yesterday. The URL in question plays fine in my browser.

Feel free to close again if I'm making some other sort of obvious user error here, but this seems like new behavior
